### PR TITLE
Use SchemaForm for the edit email modal

### PR DESCRIPTION
**Pull request by erikphansen**
Makes a new version of the edit email modal form that uses a SchemaForm component.

Also, in converting the email field to also use a SchemaForm I realized that I could reuse the entire AddressFormV2 component. So it made sense to rename and move the component, change its `address` prop to `formData` and convert it to a functional component.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by bradl3yC — APPROVED** [review]
LGTM 👍 🎉

File: src/platform/user/profile/vet360/components/EmailField/EmailField.jsx [hunk 1]
```diff
@@ -6,16 +6,43 @@ import { API_ROUTES, FIELD_NAMES } from 'vet360/constants';
...
+const formSchema = {
+  type: 'object',
+  properties: {
+    emailAddress: {
+      type: 'string',
+      // This regex was taken from the HCA but modified to allow leading and
+      // trailing whitespace to reduce false errors. The `convertDataToPayload`
+      // method will clean up the whitespace before submission
+      pattern:
+        '^(\\s)*(([^<>()\\[\\]\\\\.,;:\\s@"]+(\\.[^<>()\\[\\]\\\\.,;:\\s@"]+)*)|(".+"))@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))(\\s)*$',
```
Comment (ncksllvn) [inline src/platform/user/profile/vet360/components/EmailField/EmailField.jsx]: So this is better than `email.indexOf('@') > 0`?

**Review by ncksllvn — APPROVED** [review]

File: src/platform/user/profile/vet360/components/EmailField/EmailField.jsx (hunk 1 above)
Comment (erikphansen) [reply src/platform/user/profile/vet360/components/EmailField/EmailField.jsx]: Well it checks for the .tld part as well. It seems like a solid regex in my testing

Comment (ncksllvn) [reply src/platform/user/profile/vet360/components/EmailField/EmailField.jsx]: Completely kidding, the regex is way better I'm sure 🙂 My code would would be just fine with `_@` as an email address lol

Comment (erikphansen) [reply src/platform/user/profile/vet360/components/EmailField/EmailField.jsx]: Wow, I must be in a weird headspace to have completely missed the sarcasm 😬

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**